basalt/autograd/graph.mojo

```diff
@@ -73,44 +73,27 @@ struct Graph:
     fn op(
         inout self,
         op: OP,
-        operand_1: Symbol,
-        operand_2: Optional[Symbol] = None,
-        operand_3: Optional[Symbol] = None,
+        *operands: Symbol,
         attributes: AttributeVector = AttributeVector(),
     ) -> Symbol:
-        var res: Symbol
-        if operand_3:
-            res = Symbol(
-                self.symbol_count,
-                dtype,
-                static_result_shape(
-                    op,
-                    operand_1.shape,
-                    operand_2.value().shape,
-                    operand_3.value().shape,
-                    attributes,
-                ),
-                self.result_trainable(operand_1, operand_2.value(), operand_3.value()),
-            )
-        elif operand_2:
-            res = Symbol(
-                self.symbol_count,
-                dtype,
-                static_result_shape(
-                    op, operand_1.shape, operand_2.value().shape, attributes
-                ),
-                self.result_trainable(operand_1, operand_2.value()),
-            )
+        var res_shape: TensorShape
+        if len(operands) == 1:
+            res_shape = static_result_shape(op, operands[0].shape, attributes)
+        elif len(operands) == 2:
+            res_shape = static_result_shape(op, operands[0].shape, operands[1].shape, attributes)
+        elif len(operands) == 3:
+            res_shape = static_result_shape(op, operands[0].shape, operands[1].shape, operands[2].shape, attributes)
```

Comment:
Not really relevant for this file, but is it possible to replace the static_result_shape call signature to take in *TensorShape instead of having 3 seperate overloads? Or even have one main overload that takes in *TensorShape and calls the appropriate version of the function?

Something like

```mojo
fn static_result_shape(op: Operator, *shapes: Symbol, attributes: AttributeVector) -> TensorShape:
    if len(shapes) == 1:
        return static_result_shape(op, shapes[0].shape, attributes)
    elif len(shapes) == 2:
        ...
    # etc..
    ```